src/components/AnnotationCreator.jsx

```diff
@@ -19,27 +19,96 @@ import usePostAnnotation from '../models/annotation/usePostAnnotation';
 import Button from './Button';
 import Text from './Text';
 import StandardDialog from './StandardDialog';
-import CustomAlert from './Alert';
+import Alert from './Alert';
+import useAddEncounter from '../models/encounter/useAddEncounter';
+import useAddAnnotationsToSightingEncounter from '../models/encounter/useAddAnnotationsToSightingEncounter';
 
 function percentageToPixels(percentValue, scalar) {
   const pixelValue = 0.01 * scalar * percentValue;
   return Math.round(Math.max(pixelValue, 0));
 }
 
+async function createEncounter(sightingData, addEncounterToSighting) {
+  const copiedProperties = pick(sightingData, [
+    'time',
+    'timeSpecificity',
+    'locationId',
+  ]);
+  const sightingId = get(sightingData, 'guid');
+  const encounterGuidsBeforeCreation = map(
+    get(sightingData, 'encounters'),
+    encounter => encounter.guid,
+    [],
+  );
+  const encounterCreationResponse = await addEncounterToSighting(
+    sightingId,
+    copiedProperties,
+  );
+  const encounterGuidsAfterCreation = map(
+    get(
+      encounterCreationResponse,
+      ['response', 'data', 'encounters'],
+      [],
+    ),
+    encounter => encounter.guid,
```

Comment:
`encounter?.guid` here please

Reply:
Done.

But also, if it's in a map, wouldn't it only ever be run if encounter existed?